internal/monitors/metadata/hostmetadata/hostmetadata.go

```diff
@@ -0,0 +1,190 @@
+package hostmetadata
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/metadata/aws/ec2metadata"
+	"github.com/signalfx/golib/metadata/hostmetadata"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/metadata"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	monitorType      = "host-metadata"
+	uptimeMetricName = "gauge.sf.host-plugin_uptime"
+)
+
+// MONITOR(host-metadata): This monitor collects metadata about a host.  It is
+// required for some views in SignalFx to operate.
+//
+// ```yaml
+// monitors:
+//   - type: host-metadata
+// ```
+//
+// In containerized environments host `/etc` and `/proc` may not be located
+// directly under the root path.  You can specify the path to `proc` and `etc`
+// using the monitor configurations `procFSPath` and `etcPath`
+//
+// ```yaml
+// monitors:
+//   - type: host-metadata
+//     procFSPath: "/hostfs/proc"
+//     etcPath: "/hostfs/etc"
+// ```
+
+// GAUGE(gauge.sf.host-plugin_uptime): The time this monitor has been running in
+// milliseconds.  Dimensions include `signalfx_agent`, `collectd`,
+// `kernel_release`, `kernel_version`
+
+var logger = log.WithFields(log.Fields{"monitorType": monitorType})
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{Monitor: &metadata.Monitor{}} }, &Config{})
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `acceptsEndpoints:"false"`
```

Comment:
Set `singleInstance:"true"` as well since this should never be configured twice.